packages/@aws-cdk/aws-codecommit/lib/repository.ts

```diff
@@ -255,17 +255,27 @@ export interface RepositoryProps {
  * Provides a CodeCommit Repository
  */
 export class Repository extends RepositoryBase {
+
+  public static fromRepositoryName(scope: Construct, id: string, repositoryName: string): IRepository {
+    return Repository.fromRepositoryAttributes(scope, id, { repositoryName });
+  }
+
+  public static fromRepositoryArn(scope: Construct, id: string, repositoryArn: string): IRepository {
+    // arn:aws:codecommit:us-east-1:123456789012:MyDemoRepo
+    return Repository.fromRepositoryName(scope, id, scope.node.stack.parseArn(repositoryArn).resource);
```

Comment:
This loses region and account information that might be present in the ARN. If anything, the relationship between them should be reversed, `fromName()` should construct an ARN implicitly in the current account and region, which is then passed to `fromArn()`. Have you used this pattern in other places?

Reply:
you are right of course